src/contrib/persistence/Akka.Persistence.Sql.TestKit/SqlSnapshotConnectionFailureSpec.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.Sql.TestKit
+{
+    public abstract class SqlSnapshotConnectionFailureSpec : Akka.TestKit.Xunit2.TestKit
+    {
+        protected static readonly string DefaultInvalidConnectionString = "INVALID_CONNECTION_STRING";
+
+        public SqlSnapshotConnectionFailureSpec(Config config = null, ITestOutputHelper output = null) : base(config)
+        {
+        }
+
+        [Fact]
+        public void Persistent_actor_should_throw_exception_upon_connection_failure_when_saving_snapshot()
+        {
+            EventFilter.Exception<Exception>().ExpectOne(() =>
```

Comment:
This test expects an exception to be thrown when a `PersistenceActor` is told to take a snapshot, but the actor cannot initialize its `SqlSnapshotStore` because it's been given a bad connection string.